src/typescript/Scala.tmLanguage.ts

```diff
@@ -590,8 +590,16 @@ export const scalaTmLanguage: TmLanguage = {
       ]
     },
     'scala-quoted': {
-      match: "('\\{|'\\[)(?!')",
-      name: 'constant.other.quoted.scala'
+      patterns: [
+        { // Start of `'{ .. }` or `${ .. }`
+          match: "['$]\\{(?!')",
+          name: 'punctuation.section.block.begin.scala'
+        },
+        { // Start of `'[ .. ]`
+          match: "'\\[(?!')",
+          name: 'meta.bracket.scala'
```

Comment:
Is there a reason that `'{` and `'[` are assigned such different tokens? Also, I don't understand why `'` and `{` have different colors on my machine, even though the tests show that they should be the same token. Does this also happen on your machine, or is mine somehow misconfigured?

![image](https://user-images.githubusercontent.com/990794/102402231-c8452480-3fe4-11eb-9f78-06d7db12d86e.png)

Reply:
We need to identify those tokens separately or it would try the `Char` tokens. I chose the block as highlighting to make sure the start and end are highlighted the same way.

With this change, I am aiming to fix the inconsistency seen in GitHub.
![Screenshot 2020-12-16 at 21 31 51](https://user-images.githubusercontent.com/3648029/102403245-3f2eed00-3fe6-11eb-95d2-5294a2dfeee1.png)

I guess the tests pass correctly on your machine.

In my setup, I actually saw the old and new version in white. It is really strange that you see it differently. Are you sure you have the correct version installed locally? I remember that when trying to install a local plugin with a lower version vscode just ignored it and kept the old one.

I will experiment with different colors for vscode to see if I have something similar.

Reply:
I did fight a little with my vscode setup to make sure I had the right version installed, but I think I eventually did. 

If you want to try my colors, I'm using the "Material Theme" by Mattia Astorino, with the Material Theme High Contrast theme.

Reply:
I do get consistently the expected highlighting with different configs and on two different machines. It is probably an issue with your setup.

Reply:
Okay, in that case I'll approve and merge. Thank you for looking into it.